File: Belah Ketupat (Stroke)/C/belah-ketupat-stroke-i.c
#include <stdio.h>
int main(){
    
    int x;
    printf("Masukan Lebar Belah Ketupat: ");
    scanf("%x", &x);
    
    for(int h = 1; h <= x + 1; h++){
        printf(" ");
    }
    
    printf("*\n");
    
    for(int i = 1; i <= x; i++){
        for(int j = x; j >= i; j--){
            printf(" ");
        }
        
        for(int l = 1; l <= 1; l++){
            printf("*");
        }
        
        for(int k = 1; k <= i; k++){
            printf(" ");
        }
        
        for(int m = 2; m <= i; m++){
            printf(" ");
        }  
         
        printf("*\n");
    }
    
    printf("*");
    
    for(int v = 0; v <= x*2; v++){
        printf(" ");
    }
    
    printf("*\n");
    
    for(int p = 0; p <= x-1; p++){
        for(int q = 0; q <= p; q++){
            printf(" ");
        }
        
        printf("*");
        
        for(int s = x-1; s >= p; s--){
            printf(" ");
        }
        
        for(int t = x-2; t >= p; t--){
            printf(" ");
        }  
         
        printf("*\n");
    }
    
    for(int u = 1; u <= x+1; u++){
        printf(" ");
    }
    
    printf("*");

    return 0;
}

File: Belah Ketupat (Stroke)/GO/belah-ketupat-stroke-i.go
package main
import "fmt"

func main() {
    var x int
    fmt.Print("Masukan Lebar Belah Ketupat: ")
    fmt.Scan(&x)
    
    for h := 1; h <= x + 1; h++ {
        fmt.Print(" ");
    }
    
    fmt.Print("*\n");
    
    for i := 1; i <= x; i++ {
        for j := x; j >= i; j-- {
            fmt.Print(" ");
        }
        
        for l := 1; l <= 1; l++ {
            fmt.Print("*");
        }
        
        for k := 1; k <= i; k++ {
            fmt.Print(" ");
        }
        
        for m := 2; m <= i; m++ {
            fmt.Print(" ");
        }  
         
        fmt.Print("*\n");
    }
    
    fmt.Print("*");
    
    for v := 0; v <= x*2; v++ {
        fmt.Print(" ");
    }
    
    fmt.Print("*\n");
    
    for p := 0; p <= x-1; p++ {
        for q := 0; q <= p; q++ {
            fmt.Print(" ");
        }
        
        fmt.Print("*");
        
        for s := x-1; s >= p; s-- {
            fmt.Print(" ");
        }
        
        for t := x-2; t >= p; t-- {
            fmt.Print(" ");
        }  
         
        fmt.Print("*\n");
    }
    
    for u := 1; u <= x+1; u++ {
        fmt.Print(" ");
    }
    
    fmt.Print("*");
}
